R/design_helper_functions.R

```diff
@@ -130,6 +130,28 @@ run_design_internal.design <- function(design, current_df = NULL, results = NULL
   }
 }
 
+#' @export
+get_estimates_data <- function(design, data, start = 1, end = length(design)) {
+
+  design_stub <- design[start:end]
+  
+  estimators <- sapply(design_stub, 
+                       function(x) attr(x, "causal_type") == "estimator")
+  
+  estimators <- design_stub[estimators]
+                       
+  results_list <- list()
+  for (i in seq_along(estimators)) {
+    results_list[[i]] <- estimators[[i]](data)
+  }
```

Comment:
Rather than yet another execution path, you can set up an execution_st and delegate to run_design_internal.

Reply:
https://github.com/DeclareDesign/DeclareDesign/blob/master/R/design_helper_functions.R#L105

You just need to provide results, start and current_df.

This will also handle the (inenvitable) edge cases where people have DGP steps in between estimator steps.

Reply:
My sense is that is a good feature to have, but distinct from this one. I'd like this function to take a design and data (and optionally a start and stop) and *just* get the estimates based on that data. If there are intermediate dgp steps, this wouldn't run them and some estimates would potentially break. I think that's ok. You can subset the set of steps to run via start and stop. What do others think?

Reply:
In that case, you may as well slice the design up front, and then call run_design_internal on that thing - the whole method reduces to two lines. But it would essentially be running a different design than what the user provided.

Reply:
`run_design_internal` doesn't have a data argument; am I missing something? I've mostly implemented this, but obv. won't run given that issue.

Reply:
For historical reasons it was called current_df https://github.com/DeclareDesign/DeclareDesign/blob/master/R/design_helper_functions.R#L85

Reply:
apologies, I was looking at the wrong function. thanks.